vnext/JSI/Shared/ChakraObjectRef.h

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "jsi/jsi.h"
+
+#ifdef CHAKRACORE
+#include "ChakraCore.h"
+#else
+#ifndef USE_EDGEMODE_JSRT
+#define USE_EDGEMODE_JSRT
+#endif
+#include <jsrt.h>
+#endif
+
+#include <exception>
+#include <string>
+
+namespace Microsoft::JSI {
+
+inline void CrashUponChakraError(JsErrorCode error) {
+  if (error != JsNoError) {
+    std::terminate();
```

Comment:
>terminate [](start = 9, length = 9)

This doesn't generate Watson report AFAIK, so we will never know about such errors.

Reply:
std::terminate is the what we've using when we want to crash in react native windows.

---
In reply to: [336152226](https://github.com/microsoft/react-native-windows/pull/3445#discussion_r336152226) [](ancestors = 336152226)